tests/test_integration.py

```diff
@@ -86,6 +93,10 @@ def test_pypi_error(sampleproject_dist, monkeypatch):
 
     result = dunder_main.main()
 
+    # Workaround intermittent failure from testpypi
+    if "HTTPError: 50" in result:
```

Comment:
Nit: I'd rather use a regex. The "50" seems awkward. Not required though.
```suggestion
    if re.search(r"HTTPError: 5\d\d", result):
```